### PR TITLE
Order item logic tests

### PR DESCRIPTION
**Pull request by KacFlor**

### CODE REVIEW

File: src/main/java/com/KacFlor/ShopSpring/controller/OrderController.java [hunk 1]
```diff
@@ -39,6 +40,13 @@ public ResponseEntity<Optional<Order>> getById(@PathVariable("id") Integer id){
         return ResponseEntity.ok(order);
     }
 
+    @PreAuthorize("hasAnyAuthority('" + Role.Fields.USER + "', '" + Role.Fields.ADMIN + "')")
+    @GetMapping("/{id}/order-items")
+    public List<OrderItem> getAllOrderItemsByOrderId(@PathVariable("id") Integer id){
+        return this.orderService.getAllOrderItems(id);
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/controller/OrderController.java:46]: getAllByOrderItemsId?

File: src/test/java/com/KacFlor/ShopSpring/controller/OrderItemControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        OrderItem orderItem1 = new OrderItem("Test1", 222.0, 22.22);
+        OrderItem orderItem2 = new OrderItem("Test2", 333.0, 22.22);
+        OrderItem orderItem3 = new OrderItem("Test3", 333.0, 22.22);
+
+        when(orderItemService.getAll()).thenReturn(List.of(orderItem1, orderItem2, orderItem3));
+
+        mockMvc.perform(get("/order-items"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(content().json("[{'name':'Test1'},{'name':'Test2'},{'name':'Test3'}]"));
```
Comment (Ficzer) [inline src/test/java/com/KacFlor/ShopSpring/controller/OrderItemControllerTest.java]: Where is the rest of content? for example 222.0?

File: src/main/java/com/KacFlor/ShopSpring/controller/OrderController.java (hunk 1 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/controller/OrderController.java:46]: these are all order items by order ID

File: src/test/java/com/KacFlor/ShopSpring/controller/OrderItemControllerTest.java (hunk 2 above)
Comment (KacFlor) [reply src/test/java/com/KacFlor/ShopSpring/controller/OrderItemControllerTest.java]: done

File: src/main/java/com/KacFlor/ShopSpring/controller/OrderController.java (hunk 1 above)
Comment (Ficzer) [reply src/main/java/com/KacFlor/ShopSpring/controller/OrderController.java:46]: ok then change the name to getAllByOrderId

Comment (Ficzer) [reply src/main/java/com/KacFlor/ShopSpring/controller/OrderController.java:46]: The names of the methods need to explain exacly what the method does, remember about it and fix the method names everywhere where it is needed. https://medium.com/swlh/how-to-better-name-your-functions-and-variables-e962a4ef335b

**Review by Ficzer — APPROVED** [review]

**Pull request merged**